src/compiler/checker.ts

```diff
@@ -23858,6 +23858,9 @@ namespace ts {
                                 return narrowByInKeyword(type, name, assumeTrue);
                             }
                         }
+                        else if (isPrivateIdentifier(expr.left)) {

```

Comment:
This should probably come at the start of the switch case for `SyntaxKind.InKeyword`, otherwise we're computing `target` and `leftType` for no reason.

Reply:
done :)